### PR TITLE
Servlet

### PR DESCRIPTION
**Pull request by azhara-a**
Fix dark mode css for comments section

Implement saving and fetching comments from the DataStore with an option for user to select maximum number of comments to display

Implement a button that deletes all comments

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,19 +34,43 @@
...
 
    @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    comments.add(request.getParameter("name") + ": " + request.getParameter("comment"));
+    Entity commentEntity = new Entity("Comment");
+    long timestamp = System.currentTimeMillis();
+
+    commentEntity.setProperty("name", request.getParameter("name"));
+    commentEntity.setProperty("comment", request.getParameter("comment"));
+    commentEntity.setProperty("timestamp", timestamp);
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:70]: nit: as a good practice you could create constants for these Strings, especially the ones that are used in several places

`private static final String PROPERTY_NAME = "name";`

https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:1]: 2020

Comment (lminchillo) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:1]: Just realized that a lot of the files say 2019, perhaps do a search in the whole repo and replace with 2020

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -23,8 +23,23 @@ $('.navbar-nav>li>a').on('click', function () {
  * Fetches a random fact from the server and adds it to the DOM.
```
Comment (lminchillo) [inline portfolio/src/main/webapp/script.js:23]: Update doc

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -36,6 +51,12 @@ function getComments() {
 function createParagraph(text) {
```
Comment (lminchillo) [inline portfolio/src/main/webapp/script.js:51]: doc says <li> but this creates a <p> element?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.google.sps.data;
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:1]: Please add copyright to this file (and any other that might have been commited without it)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (azhara-a) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:70]: Got it!

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 2 above)
Comment (azhara-a) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:1]: Will do

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (azhara-a) [reply portfolio/src/main/webapp/script.js]: Got it

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (azhara-a) [reply portfolio/src/main/webapp/script.js:57]: Will fix this

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 5 above)
Comment (azhara-a) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:1]: Got it!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -23,22 +31,54 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns fun facts */
+/* Servlet that:
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Nice descriptive comment! :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -23,22 +31,54 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns fun facts */
+/* Servlet that: 
+ * in Get request returns N comments from Datastore where N is a parameter called limit;
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: Nice descriptive comment!

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Servlet that deletes all comments in the Post request */
+@WebServlet("/delete-data")
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: Hmm, does this mean anyone can make a POST request to your server and delete everyone's comments? That would not be okay in a real-life server :)

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+public class DeleteDataServlet extends HttpServlet {
+  private static final String ENTITY_COMMENT = "Comment";
+
+   @Override
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:37]: nit: remove preceding whitespace.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 8 above)
Comment (azhara-a) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: What if I add an input field password and, on a Post request, my servlet will check this password against an environment variable? If they match, the comments will be deleted, else will display a relevant message

Comment (mugdhalakhani) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: awesome, that's what I wanted to hear. :)

**Pull request merged**